assets/js/modules/analytics-4/datastore/webdatastreams.js

```diff
@@ -555,23 +555,12 @@ const baseSelectors = {
 	isLoadingWebDataStreams: createRegistrySelector(
 		( select ) =>
 			( state, { hasModuleAccess } ) => {
-				const accountID = select( MODULES_ANALYTICS ).getAccountID();
-
 				const propertyID =
 					select( MODULES_ANALYTICS_4 ).getPropertyID();
 
-				const loadedAccounts =
-					select( MODULES_ANALYTICS ).hasFinishedResolution(
-						'getAccounts'
-					);
-
-				const loadedProperties =
-					hasModuleAccess !== false
-						? select( MODULES_ANALYTICS_4 ).hasFinishedResolution(
-								'getProperties',
-								[ accountID ]
-						  )
-						: true;
+				const loadedAccounts = select(
```

Comment:
The same here. We can make this selector more efficient if we use early returns. This will let us avoid making excessive selects when we already got the information that webdatastreams are being loaded.

We can update all checks here to return early.

Reply:
Updated ✅